tests/load_straw_prep_data.py

```diff
@@ -0,0 +1,340 @@
+    
+    return cpal_list, straw_information, cpal_prefix_list
+    
+def parse_files():
+    cpal_list = []
+    straw_information = {}
+    pp_grades=['A','B','C','D']
+    problem_files=['CPAL0002.csv', 'CPAL0040.csv', 'CPAL0025.csv', 'prep_CPAL7653.csv', 'CPAL1234.csv', 'prep_CPAL1234.csv', 'CPAL0010.csv', 'CPAL0056',
+    'CPAL3945.csv', 'CPAL6997.csv', 'CPAL0484.csv', 'CPAL9119.csv', 'CPAL0312.csv', 'CPAL9999.csv', 'CPAL1551.csv', 'prep_CPAL0591.csv',
+    'prep_CPAL7474.csv', 'prep_CPAL8476.csv', 'CPAL1008.csv', 'prep_CPAL0615.csv', 'CPAL2929.csv', 'prep_CPAL0659.csv',
+    'CPAL7326.csv', 'prep_CPAL0666.csv', 'prep_CPAL0668.csv', 'prep_CPAL0669.csv', 'prep_CPAL2081.csv', 'prep_CPAL2082.csv']
+
+    cpal_list, straw_information, cpal_prefix_list = get_straws(problem_files)
+
+    return cpal_list, straw_information, cpal_prefix_list
+
+    
+    
+def update_straw_table(straw_information, cpal_prefix_list):
+    database = GetLocalDatabasePath()
+    engine = sqla.create_engine("sqlite:///" + database)
+    connection = engine.connect()
+    
+    # for all straws not present in straw table, add them
+    for i in cpal_prefix_list:
+        cpal = i['cpal']
+        for y in range(len(straw_information[cpal])):
+            straw_id = straw_information[cpal][y]['id'][2::].lstrip('0')
+            batch = straw_information[cpal][y]['batch']
+            timestamp = int(straw_information[cpal][y]['time'])
+            
+            try:
+                straw = Straw.Straw(straw_id, batch, timestamp)
+            except:
+                print('Error saving straw ' + str(straw_id) + ' on cpal ' + str(cpal))
+
+    print('done updating straw table')
+    
+def update_straw_location_table(cpal_prefix_list):
+    for i in cpal_prefix_list:
+        cpal = i['cpal']
+        cpalid = i['cpalid']
+        time = i['time']
+    
+        try:
+            straw_location = StrawLocation.query().filter(StrawLocation.location_type == 'CPAL').filter(StrawLocation.number == cpal).all()
+            if len(straw_location) == 0:
+                location = StrawLocation('CPAL',cpal,cpalid,None,False,time)
+                #location = Pallet('CPAL',cpal,cpalid,None,time)
+        except:
+            print('Error saving cpal ' + str(cpal))
+    
+    print('done updating straw location table')
+    
+def update_measurement_prep_table(cpal_prefix_list, straw_information):
+    for i in cpal_prefix_list:
+        cpal = i['cpal']
+        cpalid = i['cpalid']
+        time = i['time']
+        
+        try:
+            for y in range(len(straw_information[cpal])):
+                straw_id = straw_information[cpal][y]['id'][2::].lstrip('0')
+                batch = straw_information[cpal][y]['batch']
+                paper_pull = straw_information[cpal][y]['grade']
+                timestamp = int(straw_information[cpal][y]['time'])
+            
+                
+                procedure = Procedure.StrawProcedure(2,cpalid,cpal,timestamp)
```

Comment:
maybe this is hwere you use the code from `Session` about starting a procedure (and incidentally creating/retrieving the strawlocation)?